sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -594,7 +596,9 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]] extends Product {
     // returns null if the product type doesn't have a primary constructor, e.g. HiveFunctionWrapper
     case p: Product => try {
```

Comment:
Could you check the master branch? We limit `toJson` to limited node types after https://github.com/apache/spark/pull/14990. `shouldConvertToJson`. It sounds like the class you mentioned above is not part of the support types